### ISSUE TITLE
"dotnet publish" failed after core projects were upgraded to 2.1.6 SDK 

### ISSUE THREAD

**Issue by venendroid** [dotnet/core#2177]
"dotnet publish" failed after core projects were upgraded to 2.1.6 SDK. 
When we are able to build the projects or solution using Visual studio 2017, we don't see any issues. It is building fine with no errors. When we are running the repo for deployment using "dotnet publish", we are seeing this below error.
`**\.nuget\packages\stylecop.msbuild\5.0.0\build\StyleCop.MSBuild.targets(101,5): error MSB4062: The "StyleCopTask" task could not be loaded from the assembly \.nuget\packages\stylecop.msbuild\5.0.0\build\..\tools\StyleCop.dll. Could not load file or assembly 'System.Windows.Forms, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089'. The system cannot find the file specified. Confirm that the <UsingTask> declaration is correct, that the `assembly` and all its dependencies are available, and that the task contains a public class that implements Microsoft.Build.Framework.ITask.**`

Urgent, please help me. It is blocking our company's release.
Few more questions, I am confused, 
1. why "publish" is running style-cop? 
2. At-least Is there a way to disable stylecop on publish?

Thank you in advance.

//PS: When we publish the project using visual studio, we did not see this issue.

**Comment by venendroid** [comment]
Here is the bug which I was found. The projects which I am building or publishing doesn't have any reference of stylecop. After digging more, I found out that one of the internal project has a nuget package reference which has a dependency of stylecop reference.

To fix this issue, I removed stylecop reference completely from the internal project and published a new nuget package. And now I referenced that new nuget package to my current projects and able to PUBLISH.
